client/src/test/java/org/threadly/litesockets/client/http/HTTPClientTests.java

```diff
@@ -305,14 +303,43 @@ public void timeoutRequest() throws IOException, HTTPParsingException {
     server.start();
     final HTTPRequestBuilder hrb = new HTTPRequestBuilder(new URL("http://localhost:"+port));
     hrb.setBody(IOUtils.EMPTY_BYTEBUFFER);
-    hrb.setTimeout(500, TimeUnit.MILLISECONDS);
+    hrb.setTimeout(200, TimeUnit.MILLISECONDS);
     final HTTPClient httpClient = new HTTPClient();
     httpClient.start();
     try{
       httpClient.request(hrb.buildClientHTTPRequest());
       fail();
     } catch(HTTPParsingException e) {
       assertEquals("HTTP Timeout!", e.getMessage());
+      // below conditions may be slightly async due to future getting a result before listeners are invoked
+      new TestCondition(() -> httpClient.getRequestQueueSize() == 0).blockTillTrue(1_000);
+      new TestCondition(() -> httpClient.getInProgressSize() == 0).blockTillTrue(1_000);
+    }
+  }
+  
+  @Test
+  public void timeoutQueuedRequest() throws IOException, HTTPParsingException {
+    int port = PortUtils.findTCPPort();
+    TCPServer server = SEI.createTCPServer("localhost", port);
+    server.start();
+    final HTTPRequestBuilder hrb = new HTTPRequestBuilder(new URL("http://localhost:"+port));
+    hrb.setBody(IOUtils.EMPTY_BYTEBUFFER);
+    hrb.setTimeout(200, TimeUnit.MILLISECONDS);
+    final HTTPClient httpClient = new HTTPClient() {
+      @Override
+      protected void processQueue() {
```

Comment:
I assume you could leave this private is setup a NoThreadSocketExecuter, and just not run select on it?